### PR TITLE
Extinguisher cabinets can now be dismantled

### PR DESCRIPTION
**Pull request by ghost**
:cl:
* rscadd: Extinguisher cabinets can now be dismantled
* rscadd: Extinguisher frames can now be built from metal

### CODE REVIEW

**Comment by Orangerush** [comment]
skullyton is /our guy/

**Review by N3X15 — APPROVED** [review]

**Comment by Zth--** [comment]
((((wiki))))

**Comment by Probe1** [comment]
Wiki then merge

File: code/game/objects/items/mountable_frames/extinguisher_cabinet.dm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/obj/item/mounted/frame/extinguisher_cabinet
+	name = "extinguisher cabinet frame"
+	desc = "For easy storage of extinguisher cabinets"
```
Comment (Exxion) [inline code/game/objects/items/mountable_frames/extinguisher_cabinet.dm]: Period. Also, you're storing the cabinet?

File: code/game/objects/structures/extinguisher.dm [hunk 2]
```diff
@@ -61,4 +70,19 @@
...
+/obj/structure/extinguisher_cabinet/proc/weld(var/obj/item/weapon/weldingtool/WE, var/mob/user)
+	if(!istype(WE))
+		return
+	if(has_extinguisher)
+		to_chat(user, "<span class='notice'>There is still an extinguisher inside</span>")
+		return
+	if(!opened)
+		to_chat(user, "<span class='notice'>\The [src] needs to be open before you can dismantle it</span>")
+		return
+	if(!WE.remove_fuel(1, user))
```
Comment (Exxion) [inline code/game/objects/structures/extinguisher.dm]: Does this check if the welder is actually on?

File: code/game/objects/structures/extinguisher.dm [hunk 3]
```diff
@@ -8,6 +8,13 @@
 	var/obj/item/weapon/extinguisher/has_extinguisher = new/obj/item/weapon/extinguisher
 	var/opened = 0
 
+/obj/structure/extinguisher_cabinet/empty
+	has_extinguisher = null
+	opened = 1
+
+/obj/structure/extinguisher_cabinet
```
Comment (Exxion) [inline code/game/objects/structures/extinguisher.dm]: I think something is missing here

File: code/game/objects/items/mountable_frames/extinguisher_cabinet.dm [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+	icon = 'icons/obj/closet.dmi'
+	icon_state = "extinguisher_empty"
+	mount_reqs = list("nospace")
+
+/obj/item/mounted/frame/extinguisher_cabinet/do_build(turf/on_wall, mob/user)
+	new /obj/structure/extinguisher_cabinet/empty(get_turf(src), get_dir(user, on_wall), 1)
```
Comment (Exxion) [inline code/game/objects/items/mountable_frames/extinguisher_cabinet.dm:9]: Does the extinguisher cabinet properly set pixel offsets with `dir`?

File: code/game/objects/structures/extinguisher.dm [hunk 5]
```diff
@@ -61,4 +70,19 @@
...
-		icon_state = "extinguisher_empty"
\ No newline at end of file
+		icon_state = "extinguisher_empty"
+
+
+/obj/structure/extinguisher_cabinet/proc/weld(var/obj/item/weapon/weldingtool/WE, var/mob/user)
+	if(!istype(WE))
+		return
+	if(has_extinguisher)
+		to_chat(user, "<span class='notice'>There is still an extinguisher inside</span>")
```
Comment (Exxion) [inline code/game/objects/structures/extinguisher.dm]: Period.

File: code/game/objects/structures/extinguisher.dm [hunk 6]
```diff
@@ -61,4 +70,19 @@
...
+		return
+	if(!opened)
+		to_chat(user, "<span class='notice'>\The [src] needs to be open before you can dismantle it</span>")
```
Comment (Exxion) [inline code/game/objects/structures/extinguisher.dm]: Period.

**Review by Exxion — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/extinguisher.dm (hunk 2 above)
Comment (PJB3005) [reply code/game/objects/structures/extinguisher.dm]: It does.

File: code/game/objects/items/mountable_frames/extinguisher_cabinet.dm (hunk 1 above)
Comment (ComicIronic) [reply code/game/objects/items/mountable_frames/extinguisher_cabinet.dm]: What's the bet on extinguisher being replaced with extinguisher cabinet and this being missed?

File: code/game/objects/items/mountable_frames/extinguisher_cabinet.dm (hunk 4 above)
Comment (ghost) [reply code/game/objects/items/mountable_frames/extinguisher_cabinet.dm:9]: I'd hope so, every other frame mounting uses this

**Comment by ghost** [comment]
@Probe1 Wiki is updated.

**Comment by ghost** [comment]
Thank you @Emilc4h

**Review by Exxion — APPROVED** [review]

**Pull request merged**